### PR TITLE
CDNA1/2 data tiling

### PR DESCRIPTION
**Pull request by bjacob**
CDNA1/2 machines are going to be in use for a while, and adding data tiling support for an architecture is just a matter of populating those 3 optional fields in `TargetWgpDetails`, and the corresponding `gpu_materialize_encoding_gfx***.mlir` test adds some coverage around the intrinsics that is useful beyond data tiling.

### CODE REVIEW

File: tests/e2e/matmul/CMakeLists.txt [hunk 1]
```diff
@@ -1741,6 +1741,36 @@ iree_generated_e2e_runner_test(
     "requires-gpu-cdna3"
 )
 
+iree_generated_e2e_runner_test(
+  NAME
+    e2e_matmul_cdna3_dt_f64
```
Comment (hanhanW) [inline tests/e2e/matmul/CMakeLists.txt:1746]: I'm confused that a cdna3 test suite is added but the PR title is about CDNA1/2. Why?

**Review by hanhanW — COMMENTED** [review]
I did not check if the numbers match the spec. I like the size of the PR when we add the support for CDNA1/2 data-tiling. :)

File: tests/e2e/matmul/CMakeLists.txt (hunk 1 above)
Comment (bjacob) [reply tests/e2e/matmul/CMakeLists.txt:1746]: Sorry, that's just a mishap while slicing my diff into PRs. This was meant to be part of https://github.com/iree-org/iree/pull/19099.

Comment (hanhanW) [reply tests/e2e/matmul/CMakeLists.txt:1746]: I see, can you rebase the PR (or merge the main branch into your branch)?

Comment (bjacob) [reply tests/e2e/matmul/CMakeLists.txt:1746]: @hanhanW , if I do that, I will lose the green CI status. The bottom commit here is already in main.  Can you review based on looking at the top commit?

Comment (hanhanW) [reply tests/e2e/matmul/CMakeLists.txt:1746]: ah, I see. I did not notice that there are two commits. Reviewed and it looks good to me, thanks!

**Review by hanhanW — APPROVED** [review]

**Pull request merged**